std/algorithm/sorting.d

```diff
@@ -994,15 +994,15 @@ if (isRandomAccessRange!Range && !isInfinite!Range &&
         "r and index must be same length for makeIndex.");
     static if (IndexType.sizeof < size_t.sizeof)
     {
-        enforce(r.length <= IndexType.max, "Cannot create an index with " ~
+        enforce(r.length <= size_t(IndexType.max + 1), "Cannot create an index with " ~
```

Comment:
this doesn't work, because `1` is an `int`:

```D
auto x = size_t(uint.max + 1);
assert(x == 0);
```
`uint` certainly can be a valid index type, but would always be rejected.

I think this should be:
```D
enforce(r.length <= size_t(1) + IndexType.max, "Cannot create ...");
```